src/PostgREST/Response.hs

```diff
@@ -153,9 +153,13 @@ singleUpsertResponse ctxApiRequest@ApiRequest{iPreferences=Preferences{..}} resu
   RSStandard {..} -> do
     let
       response = gucResponse rsGucStatus rsGucHeaders
+      returnStatus =
+        case rsInserted of
+          Just i -> if i > 0 then HTTP.status201 else HTTP.status200
+          _      -> HTTP.status200
```

Comment:
I tried to debug the what's wrong by check the actual value of `i` here.
```suggestion
      returnStatus =
        case rsInserted of
          Just i -> if i > 0 then HTTP.mkStatus (fromIntegral i) "" else HTTP.status200
          _      -> HTTP.status200
```

After running the tests again, I am getting the value of `i` as `48` and `49`, which is why it always responds with `201` which is very strange. 
```
test/spec/Feature/Query/UpsertSpec.hs:400:13: 
  10) Feature.Query.UpsertSpec.UPSERT, with PUT, Updating row, ignores the Range header
       status mismatch:
         expected: 200
         but got:  48

test/spec/Feature/Query/UpsertSpec.hs:310:15: 
  3) Feature.Query.UpsertSpec.UPSERT, with PUT, Inserting row, succeeds on a partitioned table with composite pk
       status mismatch:
         expected: 201
         but got:  49

```

Reply:
Sorry for the late reply. Below is why it's giving those values: